api/v1alpha3/azuremachine_types.go

```diff
@@ -0,0 +1,136 @@
+	// Azure provider. If both the AzureCluster and the AzureMachine specify the same tag name with different values, the
+	// AzureMachine's value takes precedence.
+	// +optional
+	AdditionalTags Tags `json:"additionalTags,omitempty"`
+
+	// AllocatePublicIP allows the ability to create dynamic public ips for machines where this value is true.
+	// +optional
+	AllocatePublicIP bool `json:"allocatePublicIP,omitempty"`
+}
+
+// AzureMachineStatus defines the observed state of AzureMachine
+type AzureMachineStatus struct {
+	// Ready is true when the provider resource is ready.
+	// +optional
+	Ready bool `json:"ready"`
+
+	// Addresses contains the Azure instance associated addresses.
+	Addresses []v1.NodeAddress `json:"addresses,omitempty"`
+
+	// VMState is the provisioning state of the Azure virtual machine.
+	// +optional
+	VMState *VMState `json:"vmState,omitempty"`
+
+	// ErrorReason will be set in the event that there is a terminal problem
+	// reconciling the Machine and will contain a succinct value suitable
+	// for machine interpretation.
+	//
+	// This field should not be set for transitive errors that a controller
+	// faces that are expected to be fixed automatically over
+	// time (like service outages), but instead indicate that something is
+	// fundamentally wrong with the Machine's spec or the configuration of
+	// the controller, and that manual intervention is required. Examples
+	// of terminal errors would be invalid combinations of settings in the
+	// spec, values that are unsupported by the controller, or the
+	// responsible controller itself being critically misconfigured.
+	//
+	// Any transient errors that occur during the reconciliation of Machines
+	// can be added as events to the Machine object and/or logged in the
+	// controller's output.
+	// +optional
+	ErrorReason *errors.MachineStatusError `json:"errorReason,omitempty"`
+
+	// ErrorMessage will be set in the event that there is a terminal problem
+	// reconciling the Machine and will contain a more verbose string suitable
+	// for logging and human consumption.
+	//
+	// This field should not be set for transitive errors that a controller
+	// faces that are expected to be fixed automatically over
+	// time (like service outages), but instead indicate that something is
+	// fundamentally wrong with the Machine's spec or the configuration of
+	// the controller, and that manual intervention is required. Examples
+	// of terminal errors would be invalid combinations of settings in the
+	// spec, values that are unsupported by the controller, or the
+	// responsible controller itself being critically misconfigured.
+	//
+	// Any transient errors that occur during the reconciliation of Machines
+	// can be added as events to the Machine object and/or logged in the
+	// controller's output.
+	// +optional
+	ErrorMessage *string `json:"errorMessage,omitempty"`
```

Comment:
`ErrorMessage` was renamed to `FailureMessage` in [v1alpha3](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md#external-objects-will-need-to-rename-statuserrorreason-and-statuserrormessage).
```suggestion
	FailureMessage *string `json:"failureMessage,omitempty"`
```